From 5fdd81b4ec6956fa013b65129baf4723d2b1a657 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 20 Jan 2022 16:45:10 +0200
Subject: [PATCH] [SW-69581] i2c_s: use device struct for dev specific regs

Including device specific headers from common code is not allowed
anymore, and instead the device structures should be used.
I2C slave device code is refactored to use device structure.

Change-Id: I93dcf6626f384cbc225b9c36622a61f6c8ff769a
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index a0249e8..ac6c063 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -414,6 +414,7 @@
 	device_init_efuse();
 	device_init_uart();
 	device_init_pci();
+	device_init_i2c_s();
 }
 
 __weak void hl_set_reset_cause(uint32_t reset_cause) {}
diff --git a/app/recovery/recovery.c b/app/recovery/recovery.c
index 661f10e..f4292ef 100644
--- a/app/recovery/recovery.c
+++ b/app/recovery/recovery.c
@@ -27,6 +27,7 @@
 static void device_init(void)
 {
 	device_init_uart();
+	device_init_i2c_s();
 }
 
 void recovery_main(void)
diff --git a/arch/gaudi/arm/common.c b/arch/gaudi/arm/common.c
index f498919..5df2a68 100644
--- a/arch/gaudi/arm/common.c
+++ b/arch/gaudi/arm/common.c
@@ -15,6 +15,7 @@
 #include <psoc_spi_regs.h>
 #include <psoc_gpio3_regs.h>
 #include <psoc_uart_0_regs.h>
+#include <psoc_i2c_s_regs.h>
 #include <pcie_aux_regs.h>
 #include <pcie_core_regs.h>
 #include <pcie_wrap_regs.h>
@@ -678,3 +679,27 @@
 	pci.aux_smlh_link_up_reg = mmPCIE_AUX_SMLH_LINK_UP;
 	pci.core_hot_rst_reg = mmPCIE_CORE_HOT_RST;
 }
+
+void device_init_i2c_s(void)
+{
+	i2c_s.enable_reg = mmPSOC_I2C_S_IC_ENABLE;
+	i2c_s.sar_reg = mmPSOC_I2C_S_IC_SAR;
+	i2c_s.con_reg = mmPSOC_I2C_S_IC_CON;
+	i2c_s.intr_mask_reg = mmPSOC_I2C_S_IC_INTR_MASK;
+	i2c_s.intr_stat_reg = mmPSOC_I2C_S_IC_INTR_STAT;
+	i2c_s.rx_tl_reg = mmPSOC_I2C_S_IC_RX_TL;
+	i2c_s.tx_tl_reg = mmPSOC_I2C_S_IC_TX_TL;
+	i2c_s.data_cmd_reg = mmPSOC_I2C_S_IC_DATA_CMD;
+	i2c_s.clr_tx_abrt_reg = mmPSOC_I2C_S_IC_CLR_TX_ABRT;
+	i2c_s.clr_rx_under_reg = mmPSOC_I2C_S_IC_CLR_RX_UNDER;
+	i2c_s.clr_rx_over_reg = mmPSOC_I2C_S_IC_CLR_RX_OVER;
+	i2c_s.clr_tx_over_reg = mmPSOC_I2C_S_IC_CLR_TX_OVER;
+	i2c_s.clr_rx_done_reg = mmPSOC_I2C_S_IC_CLR_RX_DONE;
+	i2c_s.clr_activity_reg = mmPSOC_I2C_S_IC_CLR_ACTIVITY;
+	i2c_s.clr_stop_det_reg = mmPSOC_I2C_S_IC_CLR_STOP_DET;
+	i2c_s.clr_start_det_reg = mmPSOC_I2C_S_IC_CLR_START_DET;
+	i2c_s.clr_gen_call_reg = mmPSOC_I2C_S_IC_CLR_GEN_CALL;
+	i2c_s.clr_rd_req_reg = mmPSOC_I2C_S_IC_CLR_RD_REQ;
+	i2c_s.clr_intr_reg = mmPSOC_I2C_S_IC_CLR_INTR;
+	i2c_s.rxflr_reg = mmPSOC_I2C_S_IC_RXFLR;
+}
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index aa550d7..b13ce53 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -16,6 +16,7 @@
 #include <psoc_qspi_regs.h>
 #include <psoc_spi_regs.h>
 #include <psoc_efuse_regs.h>
+#include <psoc_i2c_s_regs.h>
 #include <pcie_phy_regs.h>
 #include <pcie_phy_masks.h>
 #include <pcie_wrap_regs.h>
@@ -1247,3 +1248,27 @@
 	pci.aux_smlh_link_up_reg = mmPCIE_AUX_SMLH_LINK_UP;
 	pci.core_hot_rst_reg = mmPCIE_CORE_HOT_RST;
 }
+
+void device_init_i2c_s(void)
+{
+	i2c_s.enable_reg = mmPSOC_I2C_S_IC_ENABLE;
+	i2c_s.sar_reg = mmPSOC_I2C_S_IC_SAR;
+	i2c_s.con_reg = mmPSOC_I2C_S_IC_CON;
+	i2c_s.intr_mask_reg = mmPSOC_I2C_S_IC_INTR_MASK;
+	i2c_s.intr_stat_reg = mmPSOC_I2C_S_IC_INTR_STAT;
+	i2c_s.rx_tl_reg = mmPSOC_I2C_S_IC_RX_TL;
+	i2c_s.tx_tl_reg = mmPSOC_I2C_S_IC_TX_TL;
+	i2c_s.data_cmd_reg = mmPSOC_I2C_S_IC_DATA_CMD;
+	i2c_s.clr_tx_abrt_reg = mmPSOC_I2C_S_IC_CLR_TX_ABRT;
+	i2c_s.clr_rx_under_reg = mmPSOC_I2C_S_IC_CLR_RX_UNDER;
+	i2c_s.clr_rx_over_reg = mmPSOC_I2C_S_IC_CLR_RX_OVER;
+	i2c_s.clr_tx_over_reg = mmPSOC_I2C_S_IC_CLR_TX_OVER;
+	i2c_s.clr_rx_done_reg = mmPSOC_I2C_S_IC_CLR_RX_DONE;
+	i2c_s.clr_activity_reg = mmPSOC_I2C_S_IC_CLR_ACTIVITY;
+	i2c_s.clr_stop_det_reg = mmPSOC_I2C_S_IC_CLR_STOP_DET;
+	i2c_s.clr_start_det_reg = mmPSOC_I2C_S_IC_CLR_START_DET;
+	i2c_s.clr_gen_call_reg = mmPSOC_I2C_S_IC_CLR_GEN_CALL;
+	i2c_s.clr_rd_req_reg = mmPSOC_I2C_S_IC_CLR_RD_REQ;
+	i2c_s.clr_intr_reg = mmPSOC_I2C_S_IC_CLR_INTR;
+	i2c_s.rxflr_reg = mmPSOC_I2C_S_IC_RXFLR;
+}
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index cef430d..33e726a 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -14,6 +14,7 @@
 #include <d0_psoc_efuse_regs.h>
 #include <d0_psoc_uart_0_regs.h>
 #include <d0_parc_arc0_cfg_regs.h>
+#include <d0_psoc_i2c_s_regs.h>
 #include <gaudi3/gaudi3_boot_info.h>
 
 #include <bitops.h>
@@ -174,6 +175,30 @@
 	pci.core_hot_rst_reg = mmD0_PCIE_CORE_HOT_RST;
 }
 
+void device_init_i2c_s(void)
+{
+	i2c_s.enable_reg = mmD0_PSOC_I2C_S_IC_ENABLE;
+	i2c_s.sar_reg = mmD0_PSOC_I2C_S_IC_SAR;
+	i2c_s.con_reg = mmD0_PSOC_I2C_S_IC_CON;
+	i2c_s.intr_mask_reg = mmD0_PSOC_I2C_S_IC_INTR_MASK;
+	i2c_s.intr_stat_reg = mmD0_PSOC_I2C_S_IC_INTR_STAT;
+	i2c_s.rx_tl_reg = mmD0_PSOC_I2C_S_IC_RX_TL;
+	i2c_s.tx_tl_reg = mmD0_PSOC_I2C_S_IC_TX_TL;
+	i2c_s.data_cmd_reg = mmD0_PSOC_I2C_S_IC_DATA_CMD;
+	i2c_s.clr_tx_abrt_reg = mmD0_PSOC_I2C_S_IC_CLR_TX_ABRT;
+	i2c_s.clr_rx_under_reg = mmD0_PSOC_I2C_S_IC_CLR_RX_UNDER;
+	i2c_s.clr_rx_over_reg = mmD0_PSOC_I2C_S_IC_CLR_RX_OVER;
+	i2c_s.clr_tx_over_reg = mmD0_PSOC_I2C_S_IC_CLR_TX_OVER;
+	i2c_s.clr_rx_done_reg = mmD0_PSOC_I2C_S_IC_CLR_RX_DONE;
+	i2c_s.clr_activity_reg = mmD0_PSOC_I2C_S_IC_CLR_ACTIVITY;
+	i2c_s.clr_stop_det_reg = mmD0_PSOC_I2C_S_IC_CLR_STOP_DET;
+	i2c_s.clr_start_det_reg = mmD0_PSOC_I2C_S_IC_CLR_START_DET;
+	i2c_s.clr_gen_call_reg = mmD0_PSOC_I2C_S_IC_CLR_GEN_CALL;
+	i2c_s.clr_rd_req_reg = mmD0_PSOC_I2C_S_IC_CLR_RD_REQ;
+	i2c_s.clr_intr_reg = mmD0_PSOC_I2C_S_IC_CLR_INTR;
+	i2c_s.rxflr_reg = mmD0_PSOC_I2C_S_IC_RXFLR;
+}
+
 void lkd_dev_specific_setup(void)
 {
 	cpu_boot_dev_sts_set(CPU_BOOT_DEV_STS_FW_HARD_RST_EN);
diff --git a/arch/goya/arm/common.c b/arch/goya/arm/common.c
index 060fc19..08c44a8 100644
--- a/arch/goya/arm/common.c
+++ b/arch/goya/arm/common.c
@@ -13,6 +13,7 @@
 #include <psoc_gpio3_regs.h>
 #include <psoc_spi_regs.h>
 #include <psoc_uart_0_regs.h>
+#include <psoc_i2c_s_regs.h>
 #include <pcie_aux_regs.h>
 #include <pcie_dbi_regs.h>
 #include <pcie_core_regs.h>
@@ -337,3 +338,27 @@
 	pci.aux_smlh_link_up_reg = mmPCIE_AUX_SMLH_LINK_UP;
 	pci.core_hot_rst_reg = mmPCIE_CORE_HOT_RST;
 }
+
+void device_init_i2c_s(void)
+{
+	i2c_s.enable_reg = mmPSOC_I2C_S_IC_ENABLE;
+	i2c_s.sar_reg = mmPSOC_I2C_S_IC_SAR;
+	i2c_s.con_reg = mmPSOC_I2C_S_IC_CON;
+	i2c_s.intr_mask_reg = mmPSOC_I2C_S_IC_INTR_MASK;
+	i2c_s.intr_stat_reg = mmPSOC_I2C_S_IC_INTR_STAT;
+	i2c_s.rx_tl_reg = mmPSOC_I2C_S_IC_RX_TL;
+	i2c_s.tx_tl_reg = mmPSOC_I2C_S_IC_TX_TL;
+	i2c_s.data_cmd_reg = mmPSOC_I2C_S_IC_DATA_CMD;
+	i2c_s.clr_tx_abrt_reg = mmPSOC_I2C_S_IC_CLR_TX_ABRT;
+	i2c_s.clr_rx_under_reg = mmPSOC_I2C_S_IC_CLR_RX_UNDER;
+	i2c_s.clr_rx_over_reg = mmPSOC_I2C_S_IC_CLR_RX_OVER;
+	i2c_s.clr_tx_over_reg = mmPSOC_I2C_S_IC_CLR_TX_OVER;
+	i2c_s.clr_rx_done_reg = mmPSOC_I2C_S_IC_CLR_RX_DONE;
+	i2c_s.clr_activity_reg = mmPSOC_I2C_S_IC_CLR_ACTIVITY;
+	i2c_s.clr_stop_det_reg = mmPSOC_I2C_S_IC_CLR_STOP_DET;
+	i2c_s.clr_start_det_reg = mmPSOC_I2C_S_IC_CLR_START_DET;
+	i2c_s.clr_gen_call_reg = mmPSOC_I2C_S_IC_CLR_GEN_CALL;
+	i2c_s.clr_rd_req_reg = mmPSOC_I2C_S_IC_CLR_RD_REQ;
+	i2c_s.clr_intr_reg = mmPSOC_I2C_S_IC_CLR_INTR;
+	i2c_s.rxflr_reg = mmPSOC_I2C_S_IC_RXFLR;
+}
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index ecab288..3b20a82 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -22,6 +22,7 @@
 #include <psoc_arc1_cfg_regs.h>
 #include <psoc_arc1_cfg_masks.h>
 #include <psoc_uart_0_regs.h>
+#include <psoc_i2c_s_regs.h>
 #include <pcie_core_regs.h>
 #include <pcie_phy_regs.h>
 #include <pcie_dbi_regs.h>
@@ -616,3 +617,27 @@
 	pci.aux_smlh_link_up_reg = mmPCIE_AUX_SMLH_LINK_UP;
 	pci.core_hot_rst_reg = mmPCIE_CORE_HOT_RST;
 }
+
+void device_init_i2c_s(void)
+{
+	i2c_s.enable_reg = mmPSOC_I2C_S_IC_ENABLE;
+	i2c_s.sar_reg = mmPSOC_I2C_S_IC_SAR;
+	i2c_s.con_reg = mmPSOC_I2C_S_IC_CON;
+	i2c_s.intr_mask_reg = mmPSOC_I2C_S_IC_INTR_MASK;
+	i2c_s.intr_stat_reg = mmPSOC_I2C_S_IC_INTR_STAT;
+	i2c_s.rx_tl_reg = mmPSOC_I2C_S_IC_RX_TL;
+	i2c_s.tx_tl_reg = mmPSOC_I2C_S_IC_TX_TL;
+	i2c_s.data_cmd_reg = mmPSOC_I2C_S_IC_DATA_CMD;
+	i2c_s.clr_tx_abrt_reg = mmPSOC_I2C_S_IC_CLR_TX_ABRT;
+	i2c_s.clr_rx_under_reg = mmPSOC_I2C_S_IC_CLR_RX_UNDER;
+	i2c_s.clr_rx_over_reg = mmPSOC_I2C_S_IC_CLR_RX_OVER;
+	i2c_s.clr_tx_over_reg = mmPSOC_I2C_S_IC_CLR_TX_OVER;
+	i2c_s.clr_rx_done_reg = mmPSOC_I2C_S_IC_CLR_RX_DONE;
+	i2c_s.clr_activity_reg = mmPSOC_I2C_S_IC_CLR_ACTIVITY;
+	i2c_s.clr_stop_det_reg = mmPSOC_I2C_S_IC_CLR_STOP_DET;
+	i2c_s.clr_start_det_reg = mmPSOC_I2C_S_IC_CLR_START_DET;
+	i2c_s.clr_gen_call_reg = mmPSOC_I2C_S_IC_CLR_GEN_CALL;
+	i2c_s.clr_rd_req_reg = mmPSOC_I2C_S_IC_CLR_RD_REQ;
+	i2c_s.clr_intr_reg = mmPSOC_I2C_S_IC_CLR_INTR;
+	i2c_s.rxflr_reg = mmPSOC_I2C_S_IC_RXFLR;
+}
diff --git a/include/device.h b/include/device.h
index d410703..fac2299 100644
--- a/include/device.h
+++ b/include/device.h
@@ -37,10 +37,35 @@
 	u32 core_hot_rst_reg;
 };
 
+struct device_i2c_s {
+	u32 enable_reg;
+	u32 sar_reg;
+	u32 con_reg;
+	u32 intr_mask_reg;
+	u32 intr_stat_reg;
+	u32 rx_tl_reg;
+	u32 tx_tl_reg;
+	u32 data_cmd_reg;
+	u32 clr_tx_abrt_reg;
+	u32 clr_rx_under_reg;
+	u32 clr_rx_over_reg;
+	u32 clr_tx_over_reg;
+	u32 clr_rx_done_reg;
+	u32 clr_activity_reg;
+	u32 clr_stop_det_reg;
+	u32 clr_start_det_reg;
+	u32 clr_gen_call_reg;
+	u32 clr_rd_req_reg;
+	u32 clr_intr_reg;
+	u32 rxflr_reg;
+};
+
 extern struct device_efuse efuse;
 extern struct device_uart uart;
 extern struct device_pci pci;
+extern struct device_i2c_s i2c_s;
 
 void device_init_efuse(void);
 void device_init_uart(void);
 void device_init_pci(void);
+void device_init_i2c_s(void);
diff --git a/lib/common/module_i2c_slave.c b/lib/common/module_i2c_slave.c
index 4cfef5f..543193c 100644
--- a/lib/common/module_i2c_slave.c
+++ b/lib/common/module_i2c_slave.c
@@ -8,9 +8,8 @@
 #include <stdint.h>
 #include <stddef.h>
 #include <psoc_global_conf_regs.h>
-#include <psoc_i2c_s_regs.h>
-#include <psoc_i2c_s_masks.h>
 #include <preboot.h>
+#include <device.h>
 #include <hl_fw_if.h> /* requires preboot.h for __packed */
 #include <preboot/thermal.h>
 #include <hl_i2c_slave_events.h>
@@ -42,6 +41,8 @@
 	struct i2cslave_buf buf;
 };
 
+struct device_i2c_s i2c_s;
+
 static struct i2cslave_data i2cs_data;
 
 static void i2cslave_reset_buffer(struct i2cslave_buf *buf)
@@ -56,16 +57,16 @@
 static void i2cslave_init(bool debug_mode)
 {
 	/* Check if already enabled */
-	if (regmap_readl(mmPSOC_I2C_S_IC_ENABLE)) {
+	if (regmap_readl(i2c_s.enable_reg)) {
 		preboot_puts("I2C SLAVE already initialized\n");
 		return;
 	}
 
 	if (debug_mode) {
-		regmap_writel(I2C_SLAVE_ADDRESS_DBG, mmPSOC_I2C_S_IC_SAR);
+		regmap_writel(I2C_SLAVE_ADDRESS_DBG, i2c_s.sar_reg);
 	} else {
 		/* Program IC_SAR with address to which slave responds */
-		regmap_writel(I2C_SLAVE_ADDRESS, mmPSOC_I2C_S_IC_SAR);
+		regmap_writel(I2C_SLAVE_ADDRESS, i2c_s.sar_reg);
 
 		/* Enable I2C Slave */
 		regmap_writel(0x1, mmPSOC_GLOBAL_CONF_I2C_SLV);
@@ -75,20 +76,30 @@
 	}
 
 	/* Program IC_CON */
-	regmap_writel(0x22, mmPSOC_I2C_S_IC_CON);
+	regmap_writel(0x22, i2c_s.con_reg);
 
 	/* Set INTR mask to default */
-	regmap_writel(I2C_SLAVE_DEFAULT_INTR_MASK, mmPSOC_I2C_S_IC_INTR_MASK);
+	regmap_writel(I2C_SLAVE_DEFAULT_INTR_MASK, i2c_s.intr_mask_reg);
 
 	/* Write to IC_RX_TL to set Rx FIFO threshold level */
-	regmap_writel(0x0, mmPSOC_I2C_S_IC_RX_TL);
+	regmap_writel(0x0, i2c_s.rx_tl_reg);
 
 	/* Write to IC_TX_TL to set Tx FIFO threshold level */
-	regmap_writel(0x7, mmPSOC_I2C_S_IC_TX_TL);
+	regmap_writel(0x7, i2c_s.tx_tl_reg);
 
 	/* Write 1 to IC_ENABLE to enable DW_apb_i2c */
-	regmap_writel(0x1, mmPSOC_I2C_S_IC_ENABLE);
+	regmap_writel(0x1, i2c_s.enable_reg);
 }
+
+#define PSOC_I2C_S_IC_INTR_STAT_R_RX_UNDER_MASK		0x1
+#define PSOC_I2C_S_IC_INTR_STAT_R_RX_OVER_MASK		0x2
+#define PSOC_I2C_S_IC_INTR_STAT_R_TX_OVER_MASK		0x8
+#define PSOC_I2C_S_IC_INTR_STAT_R_TX_ABRT_MASK		0x40
+#define PSOC_I2C_S_IC_INTR_STAT_R_RX_DONE_MASK		0x80
+#define PSOC_I2C_S_IC_INTR_STAT_R_ACTIVITY_MASK		0x100
+#define PSOC_I2C_S_IC_INTR_STAT_R_STOP_DET_MASK		0x200
+#define PSOC_I2C_S_IC_INTR_STAT_R_START_DET_MASK	0x400
+#define PSOC_I2C_S_IC_INTR_STAT_R_GEN_CALL_MASK		0x800
 
 static u32 i2cslave_dw_read_clear_intrbits_slave(void)
 {
@@ -106,7 +117,7 @@
 	 *
 	 * The raw version might be useful for debugging purposes.
 	 */
-	stat = regmap_readl(mmPSOC_I2C_S_IC_INTR_STAT);
+	stat = regmap_readl(i2c_s.intr_stat_reg);
 
 	/*
 	 * Do not use the IC_CLR_INTR register to clear interrupts, or
@@ -116,30 +127,33 @@
 	 * Instead, use the separately-prepared IC_CLR_* registers.
 	 */
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_TX_ABRT_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_TX_ABRT);
+		regmap_readl(i2c_s.clr_tx_abrt_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_RX_UNDER_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_RX_UNDER);
+		regmap_readl(i2c_s.clr_rx_under_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_RX_OVER_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_RX_OVER);
+		regmap_readl(i2c_s.clr_rx_over_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_TX_OVER_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_TX_OVER);
+		regmap_readl(i2c_s.clr_tx_over_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_RX_DONE_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_RX_DONE);
+		regmap_readl(i2c_s.clr_rx_done_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_ACTIVITY_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_ACTIVITY);
+		regmap_readl(i2c_s.clr_activity_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_STOP_DET_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_STOP_DET);
+		regmap_readl(i2c_s.clr_stop_det_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_START_DET_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_START_DET);
+		regmap_readl(i2c_s.clr_start_det_reg);
 	if (stat & PSOC_I2C_S_IC_INTR_STAT_R_GEN_CALL_MASK)
-		regmap_readl(mmPSOC_I2C_S_IC_CLR_GEN_CALL);
+		regmap_readl(i2c_s.clr_gen_call_reg);
 
 	return stat;
 }
 
+#define PSOC_I2C_S_IC_INTR_STAT_R_RX_FULL_MASK		0x4
+#define PSOC_I2C_S_IC_INTR_STAT_R_RD_REQ_MASK		0x20
+
 static void i2cslave_flush(void)
 {
-	u32 stat = regmap_readl(mmPSOC_I2C_S_IC_INTR_STAT);
+	u32 stat = regmap_readl(i2c_s.intr_stat_reg);
 
 	if (stat & (PSOC_I2C_S_IC_INTR_STAT_R_RX_FULL_MASK |
 		    PSOC_I2C_S_IC_INTR_STAT_R_RD_REQ_MASK))
@@ -156,7 +170,7 @@
 
 	if (!debug_mode) {
 		i2cslave_flush();
-		regmap_writel(0x1, mmPSOC_I2C_S_IC_ENABLE);
+		regmap_writel(0x1, i2c_s.enable_reg);
 		i2cslave_flush();
 	}
 
@@ -197,6 +211,8 @@
 	return NOT_FOUND;
 }
 
+#define PSOC_I2C_S_IC_CON_IC_SLAVE_DISABLE_MASK		0x40
+
 void module_i2c_s_disable(void)
 {
 	u32 val;
@@ -207,13 +223,13 @@
 		return;
 
 	i2cslave_flush();
-	regmap_writel(0x0, mmPSOC_I2C_S_IC_ENABLE);
+	regmap_writel(0x0, i2c_s.enable_reg);
 	udelay(100);
 
 	/* Write IC_SLAVE_DISABLE to IC_CON */
-	val = regmap_readl(mmPSOC_I2C_S_IC_CON);
+	val = regmap_readl(i2c_s.con_reg);
 	val |= PSOC_I2C_S_IC_CON_IC_SLAVE_DISABLE_MASK;
-	regmap_writel(val, mmPSOC_I2C_S_IC_CON);
+	regmap_writel(val, i2c_s.con_reg);
 
 	udelay(1000);
 
@@ -317,7 +333,7 @@
 
 static inline u8 *i2cslave_consume_hw(u8 *data, u8 cnt)
 {
-	register u8 val = (u8)regmap_readl(mmPSOC_I2C_S_IC_DATA_CMD);
+	register u8 val = (u8)regmap_readl(i2c_s.data_cmd_reg);
 	register u8 cmd;
 	bool is_virt;
 	bool is_pec;
@@ -392,7 +408,7 @@
 			data = i2cs_data.buf.data;
 			cnt = i2cs_data.buf.cnt;
 		}
-	} while (regmap_readl(mmPSOC_I2C_S_IC_RXFLR));
+	} while (regmap_readl(i2c_s.rxflr_reg));
 
 	i2cs_data.seq = seq;
 	i2cs_data.buf.data = data;
@@ -436,9 +452,9 @@
 
 static inline void i2cslave_produce_hw(u8 data)
 {
-	regmap_readl(mmPSOC_I2C_S_IC_CLR_RD_REQ);
-	regmap_readl(mmPSOC_I2C_S_IC_CLR_RX_UNDER);
-	regmap_writel(data, mmPSOC_I2C_S_IC_DATA_CMD);
+	regmap_readl(i2c_s.clr_rd_req_reg);
+	regmap_readl(i2c_s.clr_rx_under_reg);
+	regmap_writel(data, i2c_s.data_cmd_reg);
 }
 
 static void module_i2c_s_do_read(void)
@@ -451,7 +467,7 @@
 
 void module_i2c_s_run(void)
 {
-	u32 stat = regmap_readl(mmPSOC_I2C_S_IC_INTR_STAT);
+	u32 stat = regmap_readl(i2c_s.intr_stat_reg);
 
 	/*
 	 * Process the read request only after all the written data is
@@ -480,19 +496,19 @@
 		return;
 
 	/* Write 0 to IC_ENABLE to disable DW_apb_i2c */
-	regmap_writel(0, mmPSOC_I2C_S_IC_ENABLE);
+	regmap_writel(0, i2c_s.enable_reg);
 
 	/* wait for disable to impact */
 	udelay(100);
 
 	/* Disable all interrupts */
-	regmap_writel(0x0, mmPSOC_I2C_S_IC_INTR_MASK);
-	regmap_readl(mmPSOC_I2C_S_IC_CLR_INTR);
+	regmap_writel(0x0, i2c_s.intr_mask_reg);
+	regmap_readl(i2c_s.clr_intr_reg);
 
 	/* Write IC_SLAVE_DISABLE to IC_CON */
-	val = regmap_readl(mmPSOC_I2C_S_IC_CON);
+	val = regmap_readl(i2c_s.con_reg);
 	val |= PSOC_I2C_S_IC_CON_IC_SLAVE_DISABLE_MASK;
-	regmap_writel(val, mmPSOC_I2C_S_IC_CON);
+	regmap_writel(val, i2c_s.con_reg);
 
 	/* Make sure all transactions arrived */
 	udelay(1000);
